src/core_read_parameter_file.c

```diff
@@ -2,217 +2,260 @@
+    strncpy(ParamTag[NParam], "SupernovaRecipeOn", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->SupernovaRecipeOn);
     ParamID[NParam++] = INT;
 
-    strcpy(ParamTag[NParam], "DiskInstabilityOn");
+    strncpy(ParamTag[NParam], "DiskInstabilityOn", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->DiskInstabilityOn);
     ParamID[NParam++] = INT;
 
-    strcpy(ParamTag[NParam], "SFprescription");
+    strncpy(ParamTag[NParam], "SFprescription", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->SFprescription);
     ParamID[NParam++] = INT;
 
-    strcpy(ParamTag[NParam], "AGNrecipeOn");
+    strncpy(ParamTag[NParam], "AGNrecipeOn", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->AGNrecipeOn);
     ParamID[NParam++] = INT;
 
-    strcpy(ParamTag[NParam], "BaryonFrac");
+    strncpy(ParamTag[NParam], "BaryonFrac", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->BaryonFrac);
     ParamID[NParam++] = DOUBLE;
 
-    strcpy(ParamTag[NParam], "Omega");
+    strncpy(ParamTag[NParam], "Omega", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->Omega);
     ParamID[NParam++] = DOUBLE;
 
-    strcpy(ParamTag[NParam], "OmegaLambda");
+    strncpy(ParamTag[NParam], "OmegaLambda", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->OmegaLambda);
     ParamID[NParam++] = DOUBLE;
 
-    strcpy(ParamTag[NParam], "PartMass");
+    strncpy(ParamTag[NParam], "PartMass", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->PartMass);
     ParamID[NParam++] = DOUBLE;
 
-    strcpy(ParamTag[NParam], "BoxSize");
+    strncpy(ParamTag[NParam], "BoxSize", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->BoxSize);
     ParamID[NParam++] = DOUBLE;
 
-    strcpy(ParamTag[NParam], "EnergySN");
+    strncpy(ParamTag[NParam], "EnergySN", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->EnergySN);
     ParamID[NParam++] = DOUBLE;
 
-    strcpy(ParamTag[NParam], "EtaSN");
+    strncpy(ParamTag[NParam], "EtaSN", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->EtaSN);
     ParamID[NParam++] = DOUBLE;
 
-    strcpy(ParamTag[NParam], "Yield");
+    strncpy(ParamTag[NParam], "Yield", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->Yield);
     ParamID[NParam++] = DOUBLE;
 
-    strcpy(ParamTag[NParam], "FracZleaveDisk");
+    strncpy(ParamTag[NParam], "FracZleaveDisk", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->FracZleaveDisk);
     ParamID[NParam++] = DOUBLE;
 
-    strcpy(ParamTag[NParam], "SfrEfficiency");
+    strncpy(ParamTag[NParam], "SfrEfficiency", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->SfrEfficiency);
     ParamID[NParam++] = DOUBLE;
 
-    strcpy(ParamTag[NParam], "FeedbackReheatingEpsilon");
+    strncpy(ParamTag[NParam], "FeedbackReheatingEpsilon", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->FeedbackReheatingEpsilon);
     ParamID[NParam++] = DOUBLE;
 
-    strcpy(ParamTag[NParam], "FeedbackEjectionEfficiency");
+    strncpy(ParamTag[NParam], "FeedbackEjectionEfficiency", MAXTAGLEN);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Easily used incorrectly; doesn't always \0-terminate or check for invalid pointers [MS-banned] (CWE-120).  Risk is low because the source is a constant string.](https://app.codacy.com/manual/jacobseiler/sage-model/pullRequest?prid=4620074)